src/rgw/rgw_file.h

```diff
@@ -637,6 +657,9 @@ namespace rgw {
       DECODE_START(2, bl);
       uint32_t fh_type;
       ::decode(fh_type, bl);
+      if (fh.fh_type != fh_type &&
```

Comment:
@taoCH for clarity, could you put the subexpressions in parentheses, i.e
if ((fh.fh_type != fh_type) &&
and ff.?